cmd/clusterctl/clusterdeployer/clusterclient/clusterclient.go

```diff
@@ -52,30 +53,30 @@ type Client interface {
 	GetContextNamespace() string
 	Apply(string) error
 	Delete(string) error
+	ScaleStatefulSet(ns string, name string, scale int32) error
 	WaitForClusterV1alpha1Ready() error
 	GetClusterObjectsInNamespace(string) ([]*clusterv1.Cluster, error)
 	GetClusterObject(string, string) (*clusterv1.Cluster, error)
-	GetMachineDeploymentObjects() ([]*clusterv1.MachineDeployment, error)
 	GetMachineDeploymentObjectsInNamespace(string) ([]*clusterv1.MachineDeployment, error)
-	GetMachineSetObjects() ([]*clusterv1.MachineSet, error)
 	GetMachineSetObjectsInNamespace(string) ([]*clusterv1.MachineSet, error)
-	GetMachineObjects() ([]*clusterv1.Machine, error)
+	GetMachineSetsForMachineDeployment(*clusterv1.MachineDeployment) ([]*clusterv1.MachineSet, error)
```

Comment:
The main reason I added the 'ForMachineSet' is because I'd really like to rename `GetMachine*ObjectsInNamespace` to `GetMachine*s` which would conflict here. I'm open to alternative names, though.

Reply:
Got it, that sounds fine to me